pkg/auth/authz_handler.go

```diff
@@ -0,0 +1,139 @@
+package auth
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/openshift/assisted-service/internal/common"
+	"github.com/openshift/assisted-service/pkg/ocm"
+	"github.com/openshift/assisted-service/restapi"
+	"github.com/patrickmn/go-cache"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	bareMetalClusterResource string = "BareMetalCluster"
+	amsActionCreate          string = "create"
+	clustersPathRegex        string = "clusters/.+"
+	capabilityName           string = "bare_metal_installer_admin"
+	capabilityType           string = "Account"
+)
+
+// CreateAuthorizer returns Authorizer if auth is enabled
+func (a *AuthHandler) CreateAuthorizer() func(*http.Request) error {
+	if !a.EnableAuth {
+		return func(*http.Request) error {
+			return nil
+		}
+	}
+
+	return a.Authorizer
+}
+
+// Authorizer is used to authorize a request after the Auth function was called using the "Auth*" functions
+// and the principal was stored in the context in the "AuthKey" context value.
+func (a *AuthHandler) Authorizer(request *http.Request) error {
+	payload := PayloadFromContext(request.Context())
+	if payload == nil || payload.Username == "" {
+		// auth is disabled
+		return nil
+	}
+	username := payload.Username
+	payloadFromCache, found := a.client.Cache.Get(username)
+	if found {
+		// Update payload with values from cache
+		payloadFromCache := payloadFromCache.(*ocm.AuthPayload)
+		payload.IsUser = payloadFromCache.IsUser
+		payload.IsAdmin = payloadFromCache.IsAdmin
+	} else {
+		allowed, err := a.allowedToUseAssistedInstaller(username)
+		if err != nil {
+			logrus.Errorf("Failed to authorize user: %v", err)
```

Comment:
don't use logrus directly

Reply:
Also, dont we need to return here?

Reply:
no, the next lines are:
```
                payload.IsUser = allowed
		a.client.Cache.Set(username, payload, cache.DefaultExpiration)
```
So we are setting payload.IsUser = allowed (the value of which is probably not defined) but worth than that, we are setting the cache for this value, which means that if we could not query OCM now, for the next x minutes, we will not retry.

Reply:
right in case of an error, fixed.